File: app/src/main/java/com/chandilsachin/diettracker/AddFoodPresenter.java
package com.chandilsachin.diettracker;

import android.content.Context;
import android.content.res.AssetManager;
import android.support.annotation.NonNull;
import android.util.JsonReader;

import com.chandilsachin.diettracker.Food;
import com.chandilsachin.diettracker.io.JSONReader;
import com.ne1c.rainbowmvp.ViewState;
import com.ne1c.rainbowmvp.ViewStateListener;
import com.ne1c.rainbowmvp.base.BasePresenter;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;

/**
 * Created by Sachin Chandil on 29/04/2017.
 */

public class AddFoodPresenter extends BasePresenter<AddFoodView> implements ViewStateListener {
    public static final String TAG = AddFoodPresenter.class.getName();

    public AddFoodPresenter() {
        //addViewStateListener(this);
    }

    public ArrayList<Food> readFoodItemsFile(Context context) throws IOException {

        ArrayList<Food> list = new ArrayList<>();
        try {
            JSONArray jsonArray;
            jsonArray = JSONReader.readJsonFileToArrayFromAssets(context.getAssets(), "food_Items.json");
            for (int i = 0; i < jsonArray.length(); i++){
                JSONObject object = jsonArray.getJSONObject(i);
                JSONObject fact = object.getJSONObject("facts");
                Food food = new Food(object.getString("food_name"), object.getString("food_desc"),fact.getDouble("protein"),
                        fact.getDouble("carbs"),fact.getDouble("fat"));
                list.add(food);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return list;
    }

    public ArrayList<Food> readFoodItems(Context context) throws IOException {

        ArrayList<Food> list = DatabaseManager.getInstance(context).getAllFoodList();
        addFoodView.populateListItems(list);
        return list;
    }

    private AddFoodView addFoodView;
    @Override
    public void bindView(@NonNull AddFoodView view) {
        super.bindView(view);
        addFoodView = view;
    }

    @Override
    public void stateChanged(ViewState viewState) {

    }
}

File: app/src/main/java/com/chandilsachin/diettracker/AddFoodPresenter.kt
package com.chandilsachin.diettracker

import android.content.Context
import android.content.res.AssetManager
import android.util.JsonReader

import com.chandilsachin.diettracker.database.FoodDatabase
import com.chandilsachin.diettracker.io.JSONReader
import com.ne1c.rainbowmvp.ViewState
import com.ne1c.rainbowmvp.ViewStateListener
import com.ne1c.rainbowmvp.base.BasePresenter

import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

import java.io.IOException
import java.io.InputStream
import java.util.ArrayList

/**
 * Created by Sachin Chandil on 29/04/2017.
 */

class AddFoodPresenter {

    @Throws(IOException::class)
    fun readFoodItemsFile(context: Context): ArrayList<com.chandilsachin.diettracker.database.Food> {

        val list = ArrayList<com.chandilsachin.diettracker.database.Food>()
        try {
            val jsonArray: JSONArray
            jsonArray = JSONReader.readJsonFileToArrayFromAssets(context.assets, "food_Items.json")
            for (i in 0..jsonArray.length() - 1) {
                val `object` = jsonArray.getJSONObject(i)
                val fact = `object`.getJSONObject("facts")
                val food = com.chandilsachin.diettracker.database.Food(`object`.getString("food_name"), `object`.getString("food_desc"), fact.getDouble("protein"),
                        fact.getDouble("carbs"), fact.getDouble("fat"), fact.getDouble("calorie"))
                list.add(food)
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return list
    }

    companion object {
        val TAG = AddFoodPresenter::class.java.name
    }
}
